src/StatusQ/Controls/StatusPinInput.qml

```diff
@@ -0,0 +1,137 @@
+import QtQuick 2.0
+import StatusQ.Core.Theme 0.1
+
+Item {
+    id: root
+
+    property string pinInput: inputText.text
+    property var validator: inputText.validator
+    property int pinLen: 6
+    property int circleSpacing: 16
+    property int circleDiameter: 16
+
+    QtObject {
+        id: d
+        property int currentPinIndex: 0
+    }
+
+    function statesInitialization() {
+        d.currentPinIndex = 0
+        repeater.itemAt(d.currentPinIndex).innerState = "NEXT"
+        for (var i = 1; i < root.pinLen; i++) {
+            repeater.itemAt(i).innerState = "EMPTY"
+        }
+    }
+
+    width: (root.circleDiameter + root.circleSpacing) * root.pinLen
+    height: root.circleDiameter
+
+    Component.onCompleted: {
+        statesInitialization()
+        inputText.forceActiveFocus()
+    }
+
+    // Pin input data management object:
+    TextInput {
```

Comment:
could this be a `StatusBaseInput` or `StatusInput`?

Reply:
I'm not sure if it is necessary to use these controls as they are more complex and in the end this object is only used to manage data internally and take advantage of its validation rules (unless one of them implements more suitable or specific validation rules specific for Status, which I have not been able to see).

If we use, for example, `StatusBaseInput`, every time we instantiate a `StatusPinInput` object, we will also be creating a **Rectangle** with a **StatusIcon**, **StatusBaseText**, etc.. and all of this, only for managing data.. 

But if you can see another advantage on using them here, please let me know and, yes, for sure we can modify it!

Reply:
> But if you can see another advantage on using them here, please let me know and, yes, for sure we can modify it!

I think you're bringing good arguments that speak against using StatusBaseInput. Here's another one that would speak for it:

The validation system used for `StatusPinInput` is different from StatusQ's validation system, which is more flexible that QML's built-in solution.

I wonder if we could find a way to have StatusPinInput leverage our validation system as well, not because we need more flexibility or validation options, but because we now have inconsistent APIs.

For example, users of StatusQ will know that there's `StatusValidator` types which can be attached to StatusQ controls via `validators` property. However, this component uses `validator` and the validators are of different type/syntax as well, which can be confusing.

You'd need to keep track of which components of StatusQ don't use the StatusQ validation system.

If adding a `StatusBaseInput` here is problematic because of complexity, the only other way I see this work is to add the same validation mechanism to a simpler type like `TextInput`, if we aim for API consistency.

WDYT?

Reply:
After looking at `StatusValidator` and taking into account 2 factors (to keep API consistency and use `TextInput` to manage data), I've created the following 2 new `StatusValidator` controls with their corresponding documentation:

- `StatusRegularExpressionValidator` which wraps a QML type `RegularExpressionValidator`.

- `StatusIntValidator` which wraps a QML type `IntValidator`.

@alexandraB99 and @PascalPrecht take a look at the solution and I'll be glad to read your opinions! I hope you feel it works! :)